packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -214,10 +217,21 @@ export class ForkChoice implements IForkChoice {
           root: fromHexString(headRoot),
         });
       }
-
-      return (this.head = toBlockSummary(headNode));
+      prevHead = this.head;
+      return (this.head = newHead = toBlockSummary(headNode));
     } finally {
       if (timer) timer();
+      if (newHead && this.metrics) {
+        //if there is no prevHead, then isSameHead is false and isSameChain is true
+        const isSameHead = prevHead && toHexString(newHead.blockRoot) == toHexString(prevHead.blockRoot);
+        const isSameChain = !prevHead || isSameHead || this.isDescendant(prevHead.blockRoot, newHead.blockRoot);
+
+        if (isSameChain && !isSameHead) this.metrics.forkChoiceNewHeads.inc();
+        if (!isSameChain) this.metrics.forkChoiceNewChains.inc();
+      }
+      if (!newHead) {
+        this.metrics?.forkChoiceErrors.inc();
```

Comment:
Not finding a head is considered an error? If so why don't we throw an error? What about other thrown errors? Those won't be seen by this metric

Reply:
newHead will be assigned on if the flow reached the last line of return successfully, which is same as this.head, if there are any throws before, all of them  will be counted in.